src/h5web/dataset-visualizer/vis/MatrixVis.tsx

```diff
@@ -13,34 +21,179 @@ interface Props {
   data: HDF5Value;
 }
 
+type CellProps = Omit<GridChildComponentProps, 'data'>;
+
+interface MatrixProps {
+  children: CellRendererFunc;
+  width: number;
+  height: number;
+}
+
+interface IndexCellProps {
+  index: number;
+  style: React.CSSProperties;
+}
+
+type CellRendererFunc = (props: GridChildComponentProps) => JSX.Element;
+
 function MatrixVis(props: Props): JSX.Element {
   const { dims, data } = props;
+  const [sticky, setSticky] = useState(true);
+  const gridDimensions = {
+    rowHeight: 32,
+    rowCount: dims[0] + 1,
+    columnWidth: 116,
+    columnCount: (dims.length === 2 ? dims[1] : 1) + 1,
+  };
+  const GridSettings = createContext(gridDimensions);
+
+  const CellRenderer: CellRendererFunc = (
+    cellRendererProps: GridChildComponentProps
+  ) => {
+    const { data: itemData, rowIndex, columnIndex, style } = cellRendererProps;
+    const { NormalCellFunc } = itemData;
+    // Disable rendering of index columns (done by the innerElementType)
+    if (rowIndex * columnIndex === 0) {
+      return <></>;
+    }
+    return (
+      <NormalCellFunc
+        rowIndex={rowIndex - 1}
+        columnIndex={columnIndex - 1}
+        style={style}
+      />
+    );
+  };
 
   const accessor: AccessorFunc =
     dims.length === 1 ? row => data[row] : (row, col) => data[row][col];
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function ValueCell(cellProps: CellProps): JSX.Element {
+    const { rowIndex: row, columnIndex: col, style } = cellProps;
+    return (
+      <div
+        className={styles.cell}
+        style={style}
+        data-bg={(col + row) % 2 === 1 ? '' : undefined}
+      >
+        {numeral(accessor(row, col)).format('0.000e+0')}
+      </div>
+    );
+  }
+
+  function IndexCell(indexCellProps: IndexCellProps): JSX.Element {
+    const { index, style } = indexCellProps;
+    return (
+      <div className={styles.indexCell} style={style}>
+        {index >= 0 ? (
+          index
```

Comment:
If you remove the anchor button logic from here and introduce a generic component for `IndexRow` and `IndexColumn` as suggested in other comments, I don't think you'd need to keep `IndexCell` as a separate component.